src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.inline.hpp

```diff
@@ -312,8 +397,16 @@ ShenandoahCardCluster<RememberedSet>::block_start(const size_t card_index) const
   size_t offset = get_last_start(cur_index);
   // can avoid call via card size arithmetic below instead
   p = _rs->addr_for_card_index(cur_index) + offset;
-  // Recall that we already dealt with the co-initial object case above
+  // In the case that no object has been allocated in this card, we may not have dealt with co-initial object case above.
   assert(p < left, "obj should start before left");
+  obj = cast_to_oop(p);
+  if (p + obj->size() == left) {
+#ifdef KELVIN_DEBUG_DEEPER
+    log_info(gc)("KELVIN CONFIRMS PROBLEM");
+#endif
+    return left;
+  }
+
```

Comment:
Make this change in "fix block_start", or add an assert to prevent this use case in invocations of block_start.